middleware/instrument_test.go

```diff
@@ -18,22 +18,42 @@ import (
 
 func TestThroughputMetricHistogram(t *testing.T) {
 	tests := []struct {
-		name     string
+		testName string
 		sleep    bool
 		header   string
 		observed bool
 	}{
-		{"WithSleep", true, "unit=0, other_unit=2", true},
-		{"WithoutSleep", false, "unit=0, other_unit=2", false},
-		{"WithoutSleep", true, "", false},
-		{"WithoutSleep", false, "", false},
+		{
+			testName: "WithSleep",
+			sleep:    true,
+			header:   "unit;val=0, other_unit;val=2",
+			observed: true,
+		},
+		{
+			testName: "WithoutSleep",
+			sleep:    false,
+			header:   "unit;val=0, other_unit;val=2",
+			observed: false,
+		},
+		{
+			testName: "WithSleepEmptyHeader",
+			sleep:    true,
+			header:   "",
+			observed: false,
+		},
+		{
+			testName: "WithoutSleepEmptyHeader",
+			sleep:    false,
+			header:   "",
+			observed: false,
+		},
```

Comment:
let's add one where the header doesn't contain the unit too; that should be a common case